SmartDeviceLink/SDLMenuUpdateAlgorithm.m

```diff
@@ -0,0 +1,121 @@
+        // Set the menu status as a 1-1 array, start off will oldMenus = all Deletes, newMenu = all Adds
+        NSMutableArray<NSNumber *> *oldMenuStatus = [SDLMenuUpdateAlgorithm buildDeleteStatus:oldMenuCells];
+        NSMutableArray<NSNumber *> *newMenuStatus = [SDLMenuUpdateAlgorithm buildAddStatus:updatedMenuCells];
+
+        NSUInteger startIndex = 0;
+        for(NSUInteger oldCellIndex = run; oldCellIndex < oldMenuCells.count; oldCellIndex++) { //For each old item
+            //Create inner loop to compare old cells to new cells to find a match, if a match if found we mark the index at match for both the old and the new status to keep since we do not want to send RPCs for those cases
+            for(NSUInteger newCellIndex = startIndex; newCellIndex < updatedMenuCells.count; newCellIndex++) {
+                if([oldMenuCells[oldCellIndex] isEqual:updatedMenuCells[newCellIndex]]) {
+                    oldMenuStatus[oldCellIndex] = @(MenuCellStateKeep);
+                    newMenuStatus[newCellIndex] = @(MenuCellStateKeep);
+                    startIndex = newCellIndex + 1;
+                    break;
+                }
+            }
+        }
+        
+        NSUInteger numberOfAdds = 0;
+        for(NSUInteger status = 0; status < newMenuStatus.count; status++) {
+            // 0 = Delete   1 = Add    2 = Keep
+            if(newMenuStatus[status].integerValue == 1) {
+                numberOfAdds++;
+            }
+        }
+
+        if(numberOfAdds == 0) {
+            bestScoreMenu = [[SDLMenuRunScore alloc]  initWithOldStatus:oldMenuStatus updatedStatus:newMenuStatus score:numberOfAdds];
+            NSLog(@"BestScore: %lu", (unsigned long)bestScoreMenu.score);
+            NSLog(@"OldMenuStatus: %@",bestScoreMenu.oldStatus);
+            NSLog(@"NewMenuStatus: %@",bestScoreMenu.updatedStatus);
+            return bestScoreMenu;
+        }
+
+        NSLog(@"Run: %lu, RunScore: %lu", (unsigned long)run, (unsigned long)numberOfAdds);
+        if(bestScoreMenu == nil || numberOfAdds < bestScoreMenu.score) {
+            bestScoreMenu = [[SDLMenuRunScore alloc]  initWithOldStatus:oldMenuStatus updatedStatus:newMenuStatus score:numberOfAdds];
+        }
+    }
+
+    NSLog(@"BestScore: %lu", (unsigned long)bestScoreMenu.score);
+    NSLog(@"OldMenuStatus: %@",bestScoreMenu.oldStatus);
+    NSLog(@"NewMenuStatus: %@",bestScoreMenu.updatedStatus);
+
+    return bestScoreMenu;
+}
+
++ (NSMutableArray<NSNumber *> *)buildDeleteStatus:(NSArray<SDLMenuCell *> *)oldMenu {
+    NSMutableArray<NSNumber *> *oldMenuStatus = [[NSMutableArray alloc] init];
+    for(SDLMenuCell *cells in oldMenu) {
+        [oldMenuStatus addObject:@(MenuCellStateDelete)];
+    }
+    return [oldMenuStatus mutableCopy];
+}
+
++ (NSMutableArray<NSNumber *> *)buildAddStatus:(NSArray<SDLMenuCell *> *)newMenu {
+    NSMutableArray<NSNumber *> *newMenuStatus = [[NSMutableArray alloc] init];
+    for(SDLMenuCell *cells in newMenu) {
```

Comment:
`cells` isn't used here. Use a standard for loop to fix it.

Reply:
fixed